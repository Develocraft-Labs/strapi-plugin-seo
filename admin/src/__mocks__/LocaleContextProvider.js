import React, { createContext, useContext } from "react";

export const mockContextValue = {
  isI18nPluginInstalled: true,
  userEnabledLocales: [
    {
      code: "en",
      created_at: "2022-03-04T07:56:18.881Z",
      id: 1,
      isDefault: true,
      name: "English (en)",
      updated_at: "2022-03-04T07:56:18.881Z",
    },
  ],
  defaultLocale: "en",
};

export const LocaleContext = createContext(mockContextValue);

export const useLocaleContext = () => useContext(LocaleContext);

const LocaleContextProvider = ({ children }) => {
  return (
    <LocaleContext.Provider value={mockContextValue}>
      {children}
    </LocaleContext.Provider>
  );
};

export default LocaleContextProvider;
