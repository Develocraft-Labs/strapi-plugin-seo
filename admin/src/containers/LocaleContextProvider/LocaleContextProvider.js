import React, { createContext, useContext, useEffect, useState } from "react";
import { LoadingIndicatorPage, request } from "strapi-helper-plugin";
import getDefaultLocale from "../../utils/getDefaultLocale";
import getPlugins from "../../utils/getPlugins";

export const LocaleContext = createContext();

export const useLocaleContext = () => useContext(LocaleContext);

const LocaleContextProvider = ({ children }) => {
  const [isI18nPluginInstalled, setIsI18nPluginInstalled] = useState(null);
  const [userEnabledLocales, setUserEnabledLocales] = useState(null);
  const [defaultLocale, setDefaultLocale] = useState(null);
  const [loaded, setLoaded] = useState(false);

  const contextValue = {
    isI18nPluginInstalled,
    userEnabledLocales,
    defaultLocale,
  };

  useEffect(() => {
    const fetchUserEnabledLocales = async (isI18nPluginAvailable) => {
      if (isI18nPluginAvailable) {
        const userEnabledLocales = await request("/i18n/locales");
        const defLocale = await getDefaultLocale(userEnabledLocales);
        setDefaultLocale(defLocale);
        setUserEnabledLocales(userEnabledLocales);
      }
    };

    const isI18nInstalled = async () => {
      const { isI18nPluginAvailable } = await getPlugins();
      setIsI18nPluginInstalled(isI18nPluginAvailable);
      await fetchUserEnabledLocales(isI18nPluginAvailable);
      setLoaded(true);
    };
    isI18nInstalled();
  }, []);

  // Ensure we checked the locale and are ready to render the ui/fetch data
  if (!loaded) {
    return <LoadingIndicatorPage />;
  }

  return (
    <LocaleContext.Provider value={contextValue}>
      {children}
    </LocaleContext.Provider>
  );
};

export default LocaleContextProvider;
