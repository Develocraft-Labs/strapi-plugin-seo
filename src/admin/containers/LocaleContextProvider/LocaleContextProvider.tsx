// @ts-ignore
import { request, LoadingIndicatorPage } from '@strapi/helper-plugin';
import React, {
  createContext,
  useContext,
  useEffect,
  useState,
  useMemo,
  FC,
} from 'react';

import getDefaultLocale from '../../utils/getDefaultLocale';
import getContentTypes from '../../utils/getContentTypes';
import { ILocale } from '../../interfaces/Locales';
import ROUTES from '../../utils/routes';
import { ILocaleContext } from '../../interfaces/LocaleContext';
import isI18nStrapiPluginInstalled from '../../utils/isI18nStrapiPluginInstalled';

export const LocaleContext = createContext<ILocaleContext>({
  isI18nPluginInstalled: false,
  userEnabledLocales: [],
  defaultLocale: '',
});

export const useLocaleContext = (): ILocaleContext => useContext(LocaleContext);

const LocalContextProvider: FC = ({ children }) => {
  const [isI18nPluginInstalled, setIsI18nPluginInstalled] = useState(false);
  const [userEnabledLocales, setUserEnabledLocales] = useState<ILocale[]>([]);
  const [defaultLocale, setDefaultLocale] = useState('');
  const [loaded, setLoaded] = useState(false);

  const contextValue = useMemo(
    () => ({
      isI18nPluginInstalled,
      userEnabledLocales,
      defaultLocale,
    }),
    [isI18nPluginInstalled, userEnabledLocales, defaultLocale]
  );

  useEffect(() => {
    const fetchUserEnabledLocales = async (isI18nPluginAvailable: boolean) => {
      if (isI18nPluginAvailable) {
        const userEnabledLocalesData: ILocale[] = await request(
          ROUTES.I18N_LOCALES
        );
        const defLocale = await getDefaultLocale(userEnabledLocalesData);

        setDefaultLocale(defLocale);
        setUserEnabledLocales(userEnabledLocalesData);
      }
    };

    const isI18nInstalled = async () => {
      const { plugins } = await getContentTypes();
      const isI18nPluginAvailable = isI18nStrapiPluginInstalled(plugins);

      setIsI18nPluginInstalled(isI18nPluginAvailable);
      await fetchUserEnabledLocales(isI18nPluginAvailable);
      setLoaded(true);
    };
    isI18nInstalled();
  }, []);

  // Ensure we checked the locale and are ready to render the ui/fetch data
  if (!loaded) {
    return <LoadingIndicatorPage />;
  }

  return (
    <LocaleContext.Provider value={contextValue}>
      {children}
    </LocaleContext.Provider>
  );
};

export default LocalContextProvider;
