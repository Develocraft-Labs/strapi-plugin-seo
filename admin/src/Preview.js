import React, { useCallback, useMemo, useState } from "react";
import { useParams } from "react-router";
import { LoadingIndicator } from "strapi-helper-plugin";
import isEqual from "lodash/isEqual";
import { useLocaleContext } from "./containers/LocaleContextProvider/LocaleContextProvider";
import GooglePreview from "./components/GooglePreview/GooglePreview";
import limitString from "./utils/limitString";
import Form from "./ui/Forms/Seo";
import createRelation from "./utils/createRelation";
import { checkMetaDescriptionLength, checkTitleLength } from "./utils";
import showErrorNotification from "./utils/errorNotification";
import Resource from "./components/ResourceDetails/ResourceDetails";

import {
  JSONLD,
  METADESCRIPTION,
  TITLE,
  OGIMAGE,
  OGTITLE,
  OGTYPE,
  OGURL,
} from "./utils/constants";
import TrashCan from "./ui/Svgs/TrashCan";
import SaveButton from "./components/ui/SaveButton";
import { BoxColumn, Column, Row } from "./components/ui/common";
import styled from "styled-components";
import saveSeoToItem from "./utils/saveSeoToItem";
import { createSeo, updateSeo } from "./api/seoApi";
import getSeoCollectionTypeName from "./utils/getSeoCollectionTypeName";
import useFormValidations, { ValidatedForm } from "./hooks/useFormValidations";

const SideButton = styled.button`
  color: ${(p) => p.color || "rgb(246, 77, 10)"};
  align-items: center;
  cursor: pointer;
  display: flex;
  flex: 1;
  justify-content: normal;
  flex-flow: row nowrap;
  line-height: 22px;
  width: 100%;
  margin-left: 0.2rem;
  padding: 7px 20px;
  & + & {
    border-top: 1px solid #f6f6f6;
  }
`;

const MainContent = styled.div`
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px #e3e9f3;
  padding: 2rem;
  display: flex;
  flex-direction: column;
  position: relative;
  flex: 7;
  /**
    Hacky fix to limit width when monaco isn't resizing correctly
  */
  @media (min-width: 1439px) {
    max-width: ${(p) => (p.hasSidebar ? 70 : 100)}%;
  }
`;

const Sidebar = styled.div`
  display: flex;
  flex-direction: column;
  flex: 3;
  > * + * {
    margin-top: 2rem;
  }
`;

const Container = styled.div`
  display: flex;
  flex-direction: row-reverse;
  flex: 1;
  justify-content: space-between;

  position: relative;
  margin-left: -2rem;
  > * {
    margin-left: 2rem;
  }
  @media (max-width: 1439px) {
    flex-direction: column-reverse;
    margin-left: 0rem;
    margin-top: -2rem;

    > * {
      margin-left: 0rem;
      margin-top: 2rem;
    }
  }
`;

const fieldsConfig = {
  [TITLE]: {
    required: true,
  },
  [METADESCRIPTION]: {
    required: true,
  },
  [JSONLD]: {
    type: "json",
  },
  [OGTYPE]: {
    required: true,
  },
  [OGTITLE]: {
    required: true,
  },
  [OGIMAGE]: {
    required: true,
  },
  [OGURL]: {
    required: true,
  },
};

const PreviewHeader = styled(Row)`
  justify-content: space-between;
  padding: 1.5rem 0rem 3rem 0rem;
  margin-bottom: 2rem;
`;

const Preview = ({
  selectedSeo,
  onSave,
  resource,
  handleDeleteSeo,
  collectionTypeId,
  locale,
}) => {
  const [titleValue, setTitleValue] = useState(selectedSeo?.title || "");
  const [metaDescriptionValue, setMetaDescriptionValue] = useState(
    selectedSeo?.metaDescription || ""
  );
  const { isI18nPluginInstalled } = useLocaleContext();

  const [jsonLdValue, setJsonLdValue] = useState(selectedSeo?.jsonLd || "{}");
  const [ogTitleValue, setOgTitleValue] = useState(selectedSeo?.ogTitle || "");
  const [ogTypeValue, setOgTypeValue] = useState(selectedSeo?.ogType || "");
  const [ogUrlValue, setOgUrlValue] = useState(selectedSeo?.ogUrl || "");
  const [ogImageValue, setOgImageValue] = useState(selectedSeo?.ogImage || "");
  const [creation, setCreation] = useState(
    resource?.resourceSchema?.schema?.attributes?.seo == null
  );
  const [creationMsg, setCreationMsg] = useState("");
  const [saving, setSaving] = useState(false);
  const [active, setActive] = useState(false);
  const { uid, id } = useParams();

  const seo = useMemo(
    () => ({
      id: selectedSeo?.id,
      title: titleValue,
      metaDescription: metaDescriptionValue,
      jsonLd: jsonLdValue,
      ogType: ogTypeValue,
      ogTitle: ogTitleValue,
      ogUrl: ogUrlValue,
      ogImage: ogImageValue,
      locale: selectedSeo?.locale || locale,
      collectionTypeName:
        selectedSeo?.collectionTypeName || getSeoCollectionTypeName(resource),
      seoUid: selectedSeo?.seoUid,
    }),
    [
      selectedSeo,
      titleValue,
      metaDescriptionValue,
      jsonLdValue,
      ogTypeValue,
      ogTitleValue,
      ogUrlValue,
      ogImageValue,
      locale,
      resource,
    ]
  );

  const validation = useFormValidations({ fields: fieldsConfig, values: seo });
  const { validateForm } = validation;

  const modified = useMemo(() => {
    /*eslint no-unused-vars: ["error", { "ignoreRestSiblings": true }]*/
    if (!selectedSeo) {
      return true;
    }
    const {
      created_at,
      updated_at,
      created_by,
      updated_by,
      ...selectedSeoData
    } = selectedSeo;
    return !isEqual(selectedSeoData, seo);
  }, [selectedSeo, seo]);

  const handleSave = useCallback(
    async (e) => {
      e.preventDefault();
      const valid = validateForm(seo);
      if (!valid) {
        return;
      }
      try {
        setSaving(true);
        const response = await (seo?.id ? updateSeo(seo) : createSeo(seo));
        await saveSeoToItem({ seo: response, uid, id: collectionTypeId });

        onSave(response);
      } catch (ex) {
        showErrorNotification(ex.message);
      }
      setSaving(false);
    },
    [seo, onSave, collectionTypeId, uid, validateForm]
  );

  const handleRelation = async () => {
    await createRelation(resource.resourceSchema.schema, uid);

    setCreationMsg("Created Relation");
    setCreation(false);
  };

  const handleInputs = useCallback(
    (name, value) => {
      if (name === JSONLD) setJsonLdValue(value);
      else if (name === OGTITLE) setOgTitleValue(value);
      else if (name === OGTYPE) setOgTypeValue(value);
      else if (name === OGURL) setOgUrlValue(value);
      else if (name === OGIMAGE) setOgImageValue(value);
      else if (name === TITLE) {
        setTitleValue(value);
        checkTitleLength(titleValue);
      } else if (name === METADESCRIPTION) {
        setMetaDescriptionValue(value);
        checkMetaDescriptionLength(metaDescriptionValue);
      }
    },
    [titleValue, metaDescriptionValue]
  );

  const handleChange = useCallback(
    (e) => {
      const name = e.target.name;
      const value = e.target.value;
      setActive(true);
      handleInputs(name, value);
    },
    [handleInputs]
  );

  return (
    <Column>
      <PreviewHeader>
        <Resource resource={resource} />
        {creation ? (
          <SaveButton onClick={handleRelation}>
            {creationMsg ? creationMsg : "Create Relation"}
          </SaveButton>
        ) : (
          <SaveButton disabled={!active || !modified} onClick={handleSave}>
            {saving ? <LoadingIndicator small={true} /> : "Save"}
          </SaveButton>
        )}
      </PreviewHeader>
      <Container>
        {selectedSeo && (
          <Sidebar>
            <BoxColumn>
              <GooglePreview
                metaDescription={
                  metaDescriptionValue?.length >= 160
                    ? limitString(metaDescriptionValue, 0, 156)
                    : metaDescriptionValue
                }
                title={
                  titleValue?.length >= 70
                    ? limitString(titleValue, 0, 66)
                    : titleValue
                }
                ogUrl={ogUrlValue}
              />
            </BoxColumn>
            <BoxColumn padding={0}>
              <SideButton onClick={() => handleDeleteSeo({ id })}>
                <TrashCan />
                Delete this entry
              </SideButton>
            </BoxColumn>
          </Sidebar>
        )}
        <MainContent hasSidebar={!!selectedSeo}>
          <h3>Seo</h3>
          <Column fullWidth>
            {!!isI18nPluginInstalled && (
              <>
                <h3>Locale</h3>
                <p>{selectedSeo?.locale || locale}</p>
              </>
            )}
          </Column>
          <ValidatedForm context={validation}>
            <Form
              key={seo?.id}
              handleChange={handleChange}
              handleSubmit={handleSave}
              titleValue={titleValue}
              metaDescriptionValue={metaDescriptionValue}
              jsonLdValue={jsonLdValue}
              ogTitleValue={ogTitleValue}
              ogTypeValue={ogTypeValue}
              ogUrlValue={ogUrlValue}
              ogImageValue={ogImageValue}
            />
          </ValidatedForm>
        </MainContent>
      </Container>
    </Column>
  );
};

export default Preview;
