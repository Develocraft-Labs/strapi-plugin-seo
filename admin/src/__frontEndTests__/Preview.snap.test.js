/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";

import Preview from "../Preview";
import LocaleContextProvider from "../__mocks__/LocaleContextProvider";
import {
  previewComponentTestData,
  resourceComponentTestData,
} from "./testData";

// mocking components from strapi-helper-plugin
jest.mock("strapi-helper-plugin", () => {
  const Loader = jest.fn(({ children }) => children);
  const LoadingIndicator = jest.fn((props) => (
    <Loader
      onChangeParams={props.handleParamsChange}
      params={props.params}
      count={props.count}
    >
      <div>loading...</div>
    </Loader>
  ));

  const request = jest.fn().mockImplementationOnce(() => {});

  const Input = jest.fn((props) => (
    <input value={props.value} onChange={props.onChange} name={props.name} />
  ));
  const InputText = jest.fn((props) => <Input props={props} />);

  const TextArea = jest.fn((props) => <Input props={props} />);
  const InputTextArea = jest.fn((props) => (
    <TextArea value={props.value} onChange={props.onChange} name={props.name} />
  ));

  const JsonLd = jest.fn((props) => <Input props={props} />);
  const InputJson = jest.fn((props) => <Input props={props} />);
  const InputsIndex = jest.fn((props) => (
    <JsonLd
      type="json"
      value={props.value}
      onChange={props.onChange}
      name={props.name}
      customInputs={{ json: InputJson }}
    />
  ));
  return {
    LoadingIndicator: LoadingIndicator,
    request: request,
    InputText: InputText,
    InputTextArea: InputTextArea,
    InputsIndex: InputsIndex,
    validateInput: () => [],
    InputErrors: () => null,
  };
});

// mock useParams from react-router
jest.mock("react-router", () => ({
  useParams: jest.fn().mockReturnValue({
    id: 1,
    uid: "application::artists.artists",
  }),
}));

jest.mock("../containers/LocaleContextProvider/LocaleContextProvider.js", () =>
  require("../__mocks__/LocaleContextProvider")
);

describe("SNAP TESTING PREVIEW COMPONENT", () => {
  const { selectedSeo, refreshOptions, handleDeleteSeo } =
    previewComponentTestData;
  it("Should Render Successfully", () => {
    const component = renderer.create(
      <LocaleContextProvider>
        <Preview
          selectedSeo={selectedSeo}
          refreshOptions={refreshOptions}
          handleDeleteSeo={handleDeleteSeo}
          resource={resourceComponentTestData}
        />
      </LocaleContextProvider>
    );
    const tree = component.toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
