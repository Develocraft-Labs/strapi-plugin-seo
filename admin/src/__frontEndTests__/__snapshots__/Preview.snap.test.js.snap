// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAP TESTING PREVIEW COMPONENT Should Render Successfully 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c31 {
  padding-top: 2rem;
  margin-left: -2rem;
  margin-top: -2rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background-color: transparent;
}

.c31 > * {
  margin-left: 2rem;
  margin-top: 2rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  min-width: 300px;
}

.c3 {
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c8 {
  padding: 2rem;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px #e3e9f3;
}

.c24 {
  padding: 0;
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px #e3e9f3;
}

.c22 {
  height: 2rem;
  width: 2rem;
}

.c23 {
  color: black;
  overflow-wrap: break-word;
}

.c21 {
  color: gray;
  font-size: 1.4rem;
}

.c20 {
  color: #7676fb;
  font-size: 1.6rem;
  overflow-wrap: break-word;
}

.c19 {
  padding: 0rem 0rem 0rem 1rem;
  font-size: 1.3rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0rem 0rem 1.3rem 0rem;
  overflow-wrap: break-word;
}

.c17 {
  box-shadow: 0 0px 10px #e3e9f3;
  padding: 1rem;
  border-radius: 1rem;
  width: 290px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c13 {
  width: 1.5rem;
  height: 1.5rem;
  cursor: pointer;
}

.c14 {
  margin: 0;
  padding: 0rem 0.5rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 0rem;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
}

.c16 {
  height: 15rem;
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0rem 2rem 0rem;
}

.c10 {
  font-size: 1.5rem;
  padding: 2rem 0rem 0rem 0rem;
}

.c9 {
  font-weight: 400;
}

.c32 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-bottom: 1.8rem;
}

.c33 {
  background-color: white;
  padding-bottom: 1.8rem;
  padding-top: 1.5rem;
  width: 100%;
}

.c33 .monaco-editor span {
  font-family: "Droid Sans Mono","monospace",monospace, "Droid Sans Fallback";
}

.c33 .monaco-editor div {
  font-family: "Droid Sans Mono","monospace",monospace, "Droid Sans Fallback";
}

.c30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  background-color: white;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  width: 100%;
  margin: 0px;
  font-size: 1.3rem;
  font-weight: 400;
  line-height: normal;
  color: rgb(120,126,143);
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c27 {
  height: 1.8rem;
  width: 1.5rem;
  margin-right: 1rem;
  fill: rgb(246,77,10);
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 30px;
  padding: 0px 15px;
  font-weight: 600;
  font-size: 1.3rem;
  border-radius: 2px;
  cursor: pointer;
  outline: 0px;
  text-align: center;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 100%;
  margin-left: 1rem;
  margin-right: 0px;
  min-width: 150px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: rgb(255,255,255);
  background-color: rgb(109,187,26);
  border: 1px solid rgb(109,187,26);
}

.c5:disabled {
  color: rgb(180,182,186);
  background-color: rgb(233,234,235);
  border: 1px solid rgb(233,234,235);
}

.c26 {
  color: rgb(246,77,10);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: normal;
  -webkit-justify-content: normal;
  -ms-flex-pack: normal;
  justify-content: normal;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  line-height: 22px;
  width: 100%;
  margin-left: 0.2rem;
  padding: 7px 20px;
}

.c25 + .c25 {
  border-top: 1px solid #f6f6f6;
}

.c28 {
  background-color: #ffffff;
  border-radius: 2px;
  box-shadow: 0 2px 4px #e3e9f3;
  padding: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  -webkit-flex: 7;
  -ms-flex: 7;
  flex: 7;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 3;
  -ms-flex: 3;
  flex: 3;
}

.c7 > * + * {
  margin-top: 2rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  margin-left: -2rem;
}

.c6 > * {
  margin-left: 2rem;
}

.c2 {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 1.5rem 0rem 3rem 0rem;
  margin-bottom: 2rem;
}

@media (max-width:768px) {
  .c31 {
    margin-left: 0;
  }

  .c31 > * {
    margin-left: 0;
    min-width: 100%;
    max-width: 100%;
  }
}

@media (max-width:1440px) {
  .c6 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    margin-left: 0rem;
    margin-top: -2rem;
  }

  .c6 > * {
    margin-left: 0rem;
    margin-top: 2rem;
  }
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c0"
    >
      <div
        className="c0 c3"
      >
        <h2
          data-testid="resource-title"
        >
          Title
        </h2>
        <h3
          className="c4"
          data-testid="resource-collection-title"
        >
          API ID : Artists
        </h3>
      </div>
    </div>
    <button
      className="c5"
      disabled={true}
      onClick={[Function]}
    >
      Save
    </button>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      <div
        className="c0 c8"
      >
        <div>
          <h4
            className="c9"
          >
            Google Preview
          </h4>
          <hr />
          <h6
            className="c10"
          >
            Preview as:
          </h6>
          <div
            className="c1 c11"
            onChange={[Function]}
          >
            <div
              className="c12"
            >
              <input
                checked={true}
                className="c13"
                id="mobile"
                name="mobile"
                readOnly={true}
                type="radio"
                value="Mobile result"
              />
              <label
                className="c14"
                htmlFor="Mobile result"
              >
                Mobile result
              </label>
            </div>
            <div
              className="c15"
            >
              <input
                checked={false}
                className="c13"
                id="desktop"
                name="desktop"
                readOnly={true}
                type="radio"
                value="Desktop result"
              />
              <label
                className="c14"
                htmlFor="Desktop result"
              >
                Desktop result
              </label>
            </div>
          </div>
          <div
            className="c16"
          >
            <div
              className="c0 c17"
              data-testid="mobile-preview"
            >
              <small
                className="c18"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-globe"
                />
                 
                <span
                  className="c19"
                >
                  https://www.yourdomain.com/subpage
                </span>
              </small>
              <h3
                className="c20"
              >
                Artists Seo
              </h3>
              <div
                className="c21"
              >
                <span>
                  Mar 9, 2022 
                </span>
                <svg
                  className="c22"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    fill="gray"
                    height="0.3rem"
                    rx="2rem"
                    ry="2rem"
                    width="0.3rem"
                    x="0.6rem"
                    y="0.7rem"
                  />
                </svg>
                <span
                  className="c23"
                >
                  Artists Desc
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="c0 c24"
      >
        <button
          className="c25 c26"
          onClick={[Function]}
        >
          <svg
            className="c27"
            height="1.1rem"
            viewBox="0 0 24 24"
            width="1.1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 20a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V8h2V6h-4V4a2 2 0 0 0-2-2H9a2 2 0 0 0-2 2v2H3v2h2zM9 4h6v2H9zM8 8h9v12H7V8z"
            />
            <path
              d="M9 10h2v8H9zm4 0h2v8h-2z"
            />
          </svg>
          Delete this entry
        </button>
      </div>
    </div>
    <div
      className="c28"
    >
      <h3>
        Seo
      </h3>
      <div
        className="c29"
      >
        <h3>
          Locale
        </h3>
        <p />
      </div>
      <form
        className="c30"
        onSubmit={[Function]}
      >
        <div
          className="c1 c31"
        >
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              Title
            </h4>
            <input />
          </div>
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              Meta Description
            </h4>
            <input />
          </div>
        </div>
        <div
          className="c1 c31"
        >
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              Og Title
            </h4>
            <input />
          </div>
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              Og Type
            </h4>
            <input />
          </div>
        </div>
        <div
          className="c1 c31"
        >
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              OgUrl
            </h4>
            <input />
          </div>
          <div
            className="c0 c32"
            style={Object {}}
          >
            <h4>
              OgImage
            </h4>
            <input />
          </div>
        </div>
        <div
          className="c0 c33"
        >
          <h4>
            JsonLd
          </h4>
          <input />
        </div>
      </form>
    </div>
  </div>
</div>
`;
